src/keygen/participant.rs

```diff
@@ -13,43 +13,55 @@ use crate::{
         keygen_commit::{KeygenCommit, KeygenDecommit},
         keyshare::{KeySharePrivate, KeySharePublic},
     },
+    local_storage::{LocalStorage, TypeTag},
     messages::{KeygenMessageType, Message, MessageType},
     participant::{Broadcast, ProcessOutcome, ProtocolParticipant},
     protocol::ParticipantIdentifier,
     run_only_once,
-    storage::{PersistentStorageType, Storable, Storage},
+    storage::{PersistentStorageType, Storage},
     utils::k256_order,
     zkp::pisch::{PiSchInput, PiSchPrecommit, PiSchProof, PiSchSecret},
     CurvePoint,
 };
 use libpaillier::unknown_order::BigNumber;
 use merlin::Transcript;
 use rand::{CryptoRng, RngCore};
-use serde::{Deserialize, Serialize};
 use tracing::{info, instrument};
 
-// Storage identifiers for the keygen protocol.
-#[derive(Clone, Copy, Debug, Serialize)]
-#[serde(tag = "KeyGen")]
-enum StorageType {
-    Ready,
-    Commit,
-    Decommit,
-    SchnorrPrecom,
-    GlobalRid,
+struct StorageTypeReady;
+impl TypeTag for StorageTypeReady {
+    type Value = ();
+}
+struct StorageTypeCommit;
```

Comment:
suggestion: How do you feel about switching the order of these names, so its `CommitStorageType`? It feels more skimmable to me to have the interesting part of the name first, but I'm not sure if there's a naming convention for this.

Reply:
Yeah, I like it. `StorageType...` did feel clunky to me.